service/src/main/java/com/codeforcommunity/processor/BlockInfoProcessorImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IBlockInfoProcessor;
+import com.codeforcommunity.dto.blockInfo.BlockInfoResponse;
+import com.codeforcommunity.dto.blockInfo.BlockLeaderboardResponse;
+import com.codeforcommunity.dto.blockInfo.Individual;
+import com.codeforcommunity.dto.blockInfo.Team;
+import com.codeforcommunity.enums.BlockStatus;
+import java.util.List;
+import org.jooq.DSLContext;
+import static org.jooq.generated.Tables.BLOCK;
+import static org.jooq.generated.Tables.TEAM;
+import static org.jooq.generated.Tables.USERS;
+import static org.jooq.generated.Tables.USER_TEAM;
+import static org.jooq.impl.DSL.count;
+
+public class BlockInfoProcessorImpl implements IBlockInfoProcessor {
+  private final DSLContext db;
+
+  public BlockInfoProcessorImpl(DSLContext db) {
+    this.db = db;
+  }
+
+  @Override
+  public BlockInfoResponse getBlocks() {
+    Integer openBlocks = db.select(count(BLOCK.STATUS)).from(BLOCK)
+        .where(BLOCK.STATUS.eq(BlockStatus.OPEN)).fetchOneInto(Integer.class);
+    Integer doneBlocks = db.select(count(BLOCK.STATUS)).from(BLOCK)
+        .where(BLOCK.STATUS.eq(BlockStatus.DONE)).fetchOneInto(Integer.class);
+    Integer assignedBlocks = db.select(count(BLOCK.STATUS)).from(BLOCK)
+        .where(BLOCK.STATUS.eq(BlockStatus.RESERVED)).fetchOneInto(Integer.class);
+    return new BlockInfoResponse(doneBlocks, assignedBlocks, openBlocks);
+  }
+
+  @Override
+  public BlockLeaderboardResponse getBlockLeaderboards() {
```

Comment:
It looks like you're missing a sort in the db calls. Also, if it makes it significantly more simple, you could make the call to get the reserved and completed blocks in separate db calls.

Reply:
Yeah, so it got a lot dirtier (which wasn't the bad part lmao), but I was able to figure it out. My issue (refer to slack messages) was because I had a column named the wrong thing, [so I spent like 2 hours trying to figure that out](https://emoji.slack-edge.com/T029ZHFGL/painface/0a147899ed206327.png). (that emoji should now be part of the c4c workspace and i'm bothered that i didn't add it until now because i use it often enough.)

If you want it in separate calls, I can do that, but I will die a little on the inside.

Reply:
Put the individual call logic and the team call logic in separate helper functions, write the SQL for each of them at the top of their respective methods. If breaking it into two separate calls makes things significantly simpler then do that because this is absurd

Reply:
I refactored it into separate builder methods to hopefully make things cleaner. Let me know your thoughts.